HDFOutputer.cc

```diff
@@ -115,20 +137,59 @@ HDFOutputer::output(EventIdentifier const& iEventID,
   events_.push_back(iEventID.event);
 
   ++batch_;
-  if (batch_ == maxBatchSize_) {
+  if (total_n_events > 0) {
```

Comment:
As describe elsewhere in the review, this isn't a general enough solution as the total number of events is not actually known. Just the upper bound.